homeassistant/components/lgthinq/coordinator.py

```diff
@@ -0,0 +1,225 @@
+        self._thinq_api = thinq_api
+        self._is_connected = True
+
+        # If sub_id is NONE_KEY("_") then it should be None.
+        self._sub_id = None if sub_id == NONE_KEY else sub_id
+
+        # The device name is usually set to 'alias'.
+        # But, if the sub_id exists, it will be set to 'alias {sub_id}'.
+        # e.g. alias='MyWashTower', sub_id='dryer' then 'MyWashTower dryer'.
+        self._device_name = (
+            f"{device_api.alias} {self._sub_id}" if self._sub_id else device_api.alias
+        )
+
+        # The unique id is usually set to 'device_id'.
+        # But, if the sub_id exists, it will be set to 'device_id_{sub_id}'.
+        # e.g. device_id='TQSXXXX', sub_id='dryer' then 'TQSXXXX_dryer'.
+        self._unique_id = (
+            f"{device_api.device_id}_{self._sub_id}"
+            if self._sub_id
+            else device_api.device_id
+        )
+
+        # Get the api instance.
+        self._device_api = device_api.get_sub_device(self._sub_id) or device_api
+
+    @property
+    def thinq_api(self) -> ThinQApi:
+        """Returns the thinq api."""
+        return self._thinq_api
+
+    @property
+    def device_api(self) -> ConnectBaseDevice:
+        """Returns the device api."""
+        return self._device_api
+
+    @property
+    def device_name(self) -> str:
+        """Returns the device name."""
+        return self._device_name
+
+    @property
+    def sub_id(self) -> str | None:
+        """Returns the device sub id."""
+        return self._sub_id
+
+    @property
+    def unique_id(self) -> str:
+        """Returns the unique id."""
+        return self._unique_id
+
+    @property
+    def is_connected(self) -> bool:
+        """Check whether the device is connected or not."""
+        return self._is_connected
+
+    @is_connected.setter
+    def is_connected(self, connected: bool) -> None:
+        self._is_connected = connected
+
+    @property
+    def device_info(self) -> dr.DeviceInfo:
+        """Return the device information."""
+        return dr.DeviceInfo(
+            identifiers={(DOMAIN, self.unique_id)},
+            manufacturer=COMPANY,
+            model=self.device_api.model_name,
+            name=self.device_name,
+        )
+
+    @property
+    def tag(self) -> str:
+        """Returns the tag string."""
+        return f"[{self.device_name}]"
+
+    async def _async_update_data(self) -> dict[str, Any]:
+        """Request to the server to update the status from full response data."""
+        try:
+            data = await self.thinq_api.async_get_device_status(
+                self.device_api.device_id
+            )
+        except ThinQAPIException as exc:
+            if exc.code == ThinQAPIErrorCodes.NOT_CONNECTED_DEVICE:
+                self.is_connected = False
+            return {}
+
+        # Full response into the device api.
+        self.device_api.set_status(data)
+        self.is_connected = True
+        return data
```

Comment:
So CoordinatorEntity's are built to work together with the Coordinator. If the Coordinator raises `UpdateFailed` in `async_update_data`, it will automatically render all entities that are attached as `unavailable`, which is what we want.

If I understand the current code I think `is_connected` will be used for availability of the entities, but I think we should also be unavailable in the case when we can't successfully receive data from the device, for example when the cloud service is down or when someone's internet has problems

Reply:
Removed